### PR TITLE
Kyles branch 1

### PR DESCRIPTION
**Pull request by KyleLuoma**
Not sure how complete a branch needs to be to submit a PR. But, just for visibility I want to submit this working PR that implements the parameterized constructor in DataMatrix and introduces a test driver in Main. 

I still want to refactor some of the methods in the constructor to pare down the size of the constructor.

### CODE REVIEW

File: src/com/northvine/Main.java [hunk 1]
```diff
@@ -1,11 +1,46 @@
...
+
+/**
+ *
+ * @Author Gene Evans, Kyle Luoma, Trammel May, Trent Dehart
+ * @Date May 17th, 2017
+ * @Version 1.0.0
+ */
+public class Main
+{
+   static String[] sImageIn =
```
Comment (tramstheman) [inline src/com/northvine/Main.java]: This should probably be private unless you intend on using it in package scope.

File: src/com/northvine/reader/BarcodeImage.java [hunk 2]
```diff
@@ -1,116 +1,279 @@
...
+
+   /**
+    * Parameterized Constructor receieves a String array and converts
+    * to an imageData boolean[][] and shifts image to lower left of
+    * the array. 2D Array initialized as an array of MAX_HEIGHT and
+    * MAX_WIDTH, regardless of stringData dimensions.
+    *
+    * @param stringData 1D array of strings describing the QR image
+    */
+   public BarcodeImage(String[] stringData)
```
Comment (tramstheman) [inline src/com/northvine/reader/BarcodeImage.java]: After reviewing your solution, I think this parameter should be final.

File: src/com/northvine/reader/BarcodeImage.java [hunk 3]
```diff
@@ -1,116 +1,279 @@
...
+         for(int col = 0; col < temp.length; col++)
+         {
+            setPixel(row, col, temp[col] == DataMatrix.BLACK_CHAR); //row, col, value
+         }
+      }
+      
+      displayToConsole(); // for testing
+      
+      // Find the left
+      int leftIndex = horizontalScan();
```
Comment (tramstheman) [inline src/com/northvine/reader/BarcodeImage.java]: this should be final as we don't want it to be reassigned after finding the left index

File: src/com/northvine/reader/BarcodeImage.java [hunk 4]
```diff
@@ -1,116 +1,279 @@
...
+
+      // Find the bottom
+      int bottomIndex = verticalScan();
```
Comment (tramstheman) [inline src/com/northvine/reader/BarcodeImage.java]: this should be final as we don't want it to be reassigned after finding the bottom index

File: src/com/northvine/reader/BarcodeImage.java [hunk 5]
```diff
@@ -1,116 +1,279 @@
...
+   }
+
+   /**
+    * Private helper method repositions barcode image to lower
+    * left of imageData 2D array.
+    * 
+    * @param leftIndex X coord of vertical closed limitation line
+    * @param bottomIndex Y coord of horizontal closed limitation line
+    */
+   private void repositionImage( int leftIndex, int bottomIndex )
```
Comment (tramstheman) [inline src/com/northvine/reader/BarcodeImage.java]: There is much debate over this exactly, but generally private methods are supposed to go at the bottom of the class file. Constructors -> public -> protected -> package private -> private. 

Also there is extra white space wrapping the parameters and this does not meet the style convention standards of the class.

File: src/com/northvine/reader/BarcodeImage.java [hunk 6]
```diff
@@ -1,116 +1,279 @@
...
+      repositionImage(leftIndex, bottomIndex);
+   }
+
+   /**
+    * Private helper method repositions barcode image to lower
+    * left of imageData 2D array.
+    * 
+    * @param leftIndex X coord of vertical closed limitation line
+    * @param bottomIndex Y coord of horizontal closed limitation line
+    */
```
Comment (tramstheman) [inline src/com/northvine/reader/BarcodeImage.java]: private methods are not supposed to have Javadocs, so you can remove one of the '*'

**Multi-Line Comment**
```java
/*
my comment
*/
```

**Javadoc**

```java
/**
my javadoc
*/

File: src/com/northvine/reader/BarcodeImage.java [hunk 7]
```diff
@@ -1,116 +1,279 @@
...
+   /**
+    * Private helper method identifies the location of the horizontal
+    * closed limitation line by performing a vertical, bottom-to-top
+    * scan of imageData.
+    * 
+    * 
+    * @return y coordinate of horizontal closed limitation line in
+    *         imageData array
+    */
+   private int verticalScan()
```
Comment (tramstheman) [inline src/com/northvine/reader/BarcodeImage.java]: see my other comment on private scoped method definitions

File: src/com/northvine/reader/BarcodeImage.java [hunk 8]
```diff
@@ -1,116 +1,279 @@
...
+   /**
+    * Private helper method identifies the location of the vertical
+    * closed limitation line by performing a horizontal, left-to-right
+    * scan of imageData.
+    * 
+    * 
+    * @return x coordinate of vertical closed limitation line in
+    *         imageData array
+    */
+   private int horizontalScan()
```
Comment (tramstheman) [inline src/com/northvine/reader/BarcodeImage.java]: see my other comment on private scoped method definitions

**Comment by KyleLuoma** [comment]
Code works as intended. I will make your suggested changes. You can run the test driver in Main() when you get a chance and the imageData will be displayed in the console.

**Pull request merged**